backend/src/test/java/codezap/template/service/facade/TemplateApplicationServiceTest.java

```diff
@@ -0,0 +1,266 @@
+        }
+    }
+
+
+    @Nested
+    class FindAllBy {
+
+        @ParameterizedTest
+        @MethodSource
+        @DisplayName("사용자ID, 검색어, 카테고리ID, 태그ID로 템플릿 검색 성공")
+        void findAllBy(
+                Long memberId,
+                String keyword,
+                Long categoryId,
+                List<Long> tagIds,
+                Pageable pageable
+        ) {
+            // given
+            var member = memberRepository.save(MemberFixture.getFirstMember());
+            var category = categoryRepository.save(Category.createDefaultCategory(member));
+            var template1 = templateRepository.save(new Template(member, "title1", "description", category));
+            var sourceCode1 = sourceCodeRepository.save(new SourceCode(template1, "filename1", "content", 1));
+            thumbnailRepository.save(new Thumbnail(template1, sourceCode1));
+
+            // when & then
+            assertThatCode(() -> sut.findAllBy(memberId, keyword, categoryId, tagIds, pageable))
+                    .doesNotThrowAnyException();
+        }
+
+        static Stream<Arguments> findAllBy() {
+            return Stream.of(
+                    Arguments.of(null, null, null, null, Pageable.ofSize(1)),
+                    Arguments.of(1L, null, null, null, Pageable.ofSize(1)),
+                    Arguments.of(null, "keyword", null, null, Pageable.ofSize(1)),
+                    Arguments.of(null, null, 1L, null, Pageable.ofSize(1)),
+                    Arguments.of(null, null, null, List.of(1L, 2L), Pageable.ofSize(1)),
+                    Arguments.of(null, null, null, null, Pageable.ofSize(2))
+            );
+        }
+    }
+
+    @Nested
+    class Update {
+
+        @Test
+        @DisplayName("템플릿 업데이트 성공")
+        void update() {
+            // given
+            var member = memberRepository.save(MemberFixture.getFirstMember());
+            var category = categoryRepository.save(Category.createDefaultCategory(member));
+            var template = templateRepository.save(new Template(member, "title1", "description", category));
+            var sourceCode1 = sourceCodeRepository.save(new SourceCode(template, "filename1", "content1", 1));
+            var sourceCode2 = sourceCodeRepository.save(new SourceCode(template, "filename2", "content2", 2));
+            thumbnailRepository.save(new Thumbnail(template, sourceCode1));
+            var createRequest = List.of(new CreateSourceCodeRequest("filename3", "content3", 3));
+            var updateRequest1 = new UpdateSourceCodeRequest(
+                    sourceCode1.getId(),
+                    sourceCode1.getFilename(),
+                    sourceCode1.getContent(),
+                    sourceCode1.getOrdinal());
+            var updateRequest2 = new UpdateSourceCodeRequest(
+                    sourceCode2.getId(),
+                    sourceCode2.getFilename(),
+                    sourceCode2.getContent(),
+                    sourceCode2.getOrdinal());
```

Comment:
⚡️ 소스 코드를 저장하고 저장한 소스코드에 대한 UpdateSourceCodeRequest를 반환하는 메서드를 추출해보면 어떨까요?

Reply:
```suggestion
            var sourceCode1 = sourceCodeRepository.save(new SourceCode(template, "filename1", "content1", 1));
            var sourceCode2 = sourceCodeRepository.save(new SourceCode(template, "filename2", "content2", 2));
            thumbnailRepository.save(new Thumbnail(template, sourceCode1));
            var createRequest = List.of(new CreateSourceCodeRequest("filename3", "content3", 3));
            var updateRequest1 = updateSourceCodeRequest(sourceCode1);
            var updateRequest2 = updateSourceCodeRequest(sourceCode2);
                    
        private UpdateSourceCodeRequest updateSourceCodeRequest(SourceCode sourceCode) {
            return new UpdateSourceCodeRequest(
                    sourceCode.getId(),
                    sourceCode.getFilename(),
                    sourceCode.getContent(),
                    sourceCode.getOrdinal());
        }
```

소스코드를 저장하는데 필요한 도메인 객체들이 단언문에서 사용되고 중간에 다른 도메인을 저장하기도 하기 때문에 소스코드의 저장부터 request 반환까지 한 번에 처리하는 로직을 작성하는데는 어려움이 있어요. 

일단 updateSourceCodeRequest를 줄이기 위한 메서드 분리까지는 했습니다!